golemfactory/yagna
# NGNT driver: validate allocation

Implemented ValidateAllocation endpoint in NGNT driver.
Added an example for testing.

----
Testing instructions:

1) Run the `payment_api` example with either `--driver=zksync` or `--driver=ngnt`.

2) Run the `validate_allocation` example.
```
$ cargo run --example validate_allocation
[2020-11-17T16:40:05Z INFO  ya_service_bus::remote_router] trying to connect to: 127.0.0.1:7464
[2020-11-17T16:40:05Z INFO  ya_service_bus::connection] started connection to gsb
[2020-11-17T16:40:05Z INFO  validate_allocation] Requestor balance: 1000
[2020-11-17T16:40:05Z INFO  validate_allocation] Attempting to create allocation with invalid address...
[2020-11-17T16:40:05Z INFO  validate_allocation] Failed to create allocation (as expected).
[2020-11-17T16:40:05Z INFO  validate_allocation] Creating allocation for 500 NGNT...
[2020-11-17T16:40:05Z INFO  validate_allocation] Allocation created.
[2020-11-17T16:40:05Z INFO  validate_allocation] Creating another allocation for 500 NGNT...
[2020-11-17T16:40:05Z INFO  validate_allocation] Allocation created.
[2020-11-17T16:40:05Z INFO  validate_allocation] Attempting to create another allocation for 500 NGNT...
[2020-11-17T16:40:05Z INFO  validate_allocation] Failed to create allocation (as expected).
[2020-11-17T16:40:05Z INFO  validate_allocation] Releasing an allocation...
[2020-11-17T16:40:05Z INFO  validate_allocation] Allocation released.
[2020-11-17T16:40:05Z INFO  validate_allocation] Creating another allocation for 500 NGNT...
[2020-11-17T16:40:05Z INFO  validate_allocation] Allocation created.

```
